### DISCUSSION TITLE
Network is unreachable when using domain

### DISCUSSION THREAD

**Discussion by juancresc** [apache/kvrocks#1178] [Q&A]
Hi!

I was setting up a new instance and getting a lot of these:

E20221212 16:34:30.149008  8902 replication.cc:251] [replication] Failed to connect the master, err: Network is unreachable

The weird thing is that I was able to connect to the master using the same port and domain using the redis-cli, so it was reachable. The issue was solved when I replaced the domain name by the host IP. I wonder why kvrocks is it not able to connect but the redis-cli is

**Comment by juancresc** [comment]
adding a though here, is the a domain cache at the app level or something?

**Reply by git-hulk** [reply]
Kvrocks won't cache the domain, it totally depends on OS behavior. I'm wondering if it's related to ipv6.

**Reply by juancresc** [reply]
do we have a way to disable it in kvrocks? 
it's strange that redis-cli is working

**Reply by git-hulk** [reply]
You can use `nslookup {YOUR DOMAIN}` to have a look at if it's an ipv6 address.

**Reply by juancresc** [reply]
seems to be ipv4

```
Server:         10.1.0.2
Address:        10.1.0.2#53

Non-authoritative answer:
Name:   kvrocks-master.mydomain.com.internal
Address: 10.1.0.34

```

**Reply by juancresc** [reply]
interestingly I was hitting into this issue again. The solution was to move out of unstable to 2.2 and it's gone
@git-hulk

**Reply by git-hulk** [reply]
Thanks for @juancrescente feedback, I will investigate it soon.

**Reply by git-hulk** [reply]
@juancrescente Wanna confirm if your Kvrocks contains this commit: https://github.com/apache/incubator-kvrocks/pull/1172? if yes, it's caused by `SockConnect` with timeout implementation that doesn't support resolving domain yet. cc @caipengbo
